sycl/doc/extensions/GroupMask/GroupMask.asciidoc

```diff
@@ -259,6 +257,7 @@ struct group_mask {
   };
 
   static constexpr size_t max_bits = /* implementation-defined */;
+  static constexpr size_t marray_size = max_bits/sizeof(uint32_t)/CHAR_BIT;
```

Comment:
This is only correct if `max_bits` is evenly divisible by `sizeof(uint32_t)/CHAR_BIT`.  The spec does not guarantee that this is the case.  You could update the math here to round up in the case when it is not evenly divisible, or you could replace the math with a comment explaining the intent.

Also, there should be a description of `marray_size` above in the section "Group Masks".  Perhaps add a new sentence after this one:

> The maximum number of bits that can be stored in a group_mask is exposed as a static member variable, `group_mask::max_bits`.

Or, create a new table defining the two constexpr member variables.

Reply:
marray_size is implementation details and will be hidden from a user.
The code was updated to round up.

Reply:
If `marray_size` is an implementation detail, it cannot be part of the public API as you have now.  So, if you intend it to be an implementation detail, you should change the public member functions to not reference it, and you can delete the definition in the sample header.  There is no need to document something that is an implementation detail.

Reply:
I still think this variable needs to be defined in the spec since it is part of the public API.  I'd suggest adding a new table before the one that describes the member functions:

```
|===
|Member Variable|Description
|`static constexpr size_t max_bits`
|The maximum number of bits that can be stored in the `group_mask`. 

|`static constexpr size_t marray_size`
|The number of `uint32_t` elements in an `marray` which can represent all of
 the bits in the `group_mask`.

|===
```

And then delete the existing sentence that describes `max_bits`.

Reply:
Hm.  I don't think I gave this enough thought when setting the default argument.  Originally all of the masks were 128 bits, and so `marray<uint32_t, 4>` was universal; now we have to worry about rounding, things aren't quite as nice.

There might even be cases where this default argument would be wasteful -- for example, if `max_bits` is 8, then every representation of the mask as an `marray<uint32_t, 1>` is going to waste 24 bits.

Maybe it would be better to make this an `marray<char, 1>` instead, and say that `max_bits` must always be a multiple of `CHAR_BIT`?

Reply:
That sounds reasonable to me.